samples/csharp_dotnetcore/16.proactive-messages/Startup.cs

```diff
@@ -53,32 +58,67 @@ public void ConfigureServices(IServiceCollection services)
+                {
+                    throw new InvalidOperationException($"The .bot file does not contain an endpoint with name '{environment}'.");
+                }
+
+                options.CredentialProvider = new SimpleCredentialProvider(endpointService.AppId, endpointService.AppPassword);
+
+                // Creates a logger for the application to use.
+                ILogger logger = _loggerFactory.CreateLogger<ProactiveBot>();
+
+                // Catches any errors that occur during a conversation turn and logs them.
                 options.OnTurnError = async (context, exception) =>
                 {
-                    await context.TraceActivityAsync("Proactive bot exception", exception);
-                    await context.SendActivityAsync("Sorry, it looks like something went wrong!");
+                    logger.LogError($"Exception caught : {exception}");
+                    await context.SendActivityAsync("Sorry, it looks like something went wrong.");
                 };
 
                 // The Memory Storage used here is for local bot debugging only. When the bot
                 // is restarted, everything stored in memory will be gone.
                 IStorage dataStore = new MemoryStorage();
 
+                // For production bots use the Azure Blob or
+                // Azure CosmosDB storage providers. For the Azure
+                // based storage providers, add the Microsoft.Bot.Builder.Azure
+                // Nuget package to your solution. That package is found at:
+                // https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/
+                // Uncomment the following lines to use Azure Blob Storage
+                // //Storage configuration name or ID from the .bot file.
+                // const string StorageConfigurationId = "<STORAGE-NAME-OR-ID-FROM-BOT-FILE>";
+                // var blobConfig = botConfig.FindServiceByNameOrId(StorageConfigurationId);
+                // if (!(blobConfig is BlobStorageService blobStorageConfig))
+                // {
+                //    throw new InvalidOperationException($"The .bot file does not contain an blob storage with name '{StorageConfigurationId}'.");
+                // }
+                // // Default container name.
+                // const string DefaultBotContainer = "<DEFAULT-CONTAINER>";
+                // var storageContainer = string.IsNullOrWhiteSpace(blobStorageConfig.Container) ? DefaultBotContainer : blobStorageConfig.Container;
+                // IStorage dataStore = new Microsoft.Bot.Builder.Azure.AzureBlobStorage(blobStorageConfig.ConnectionString, storageContainer);
+
                 // Create Job State object.
                 // The Job State object is where we persist anything at the job-scope.
                 // It's independent of any user or conversation.
                 var jobState = new JobState(dataStore);
                 options.State.Add(jobState);
```

Comment:
>options.State.Add(jobState); [](start = 16, length = 28)

Instead of adding JobState to options.State, do a services.AddSingleton(jobState)